countComponents.js

```diff
@@ -0,0 +1,75 @@
+/*
+graph of nodes
+
+array[i] = [a, b]
+
+i: n of nodes
+o: => number of connected components
+
+Input: n = 5, edges = [[0,1],[1,2],[3,4]]
+Output: 2
+
+0 - 1 - 2
+
+3 - 4
+
+*/
+
+
+const buildGraph = (edges) => {
+    const graph = {};
+
+    for (const edge of edges) {
+        const [nodeOne, nodeTwo] = edge;
+
+        const isNodeOne = graph.hasOwnProperty(nodeOne);
+        const isNodeTwo = graph.hasOwnProperty(nodeTwo);
+
+        if (!isNodeOne) graph[nodeOne] = [];
+        if (!isNodeTwo) graph[nodeTwo] = [];
+
+        graph[nodeOne].push(nodeTwo);
+        graph[nodeTwo].push(nodeOne);
+    }
+
+    return graph;
+};
+
+const markAsVisited = (node, edges, visited) => {
+    const queue = new Queue();
+    queue.enqueue([node]);
+
+    const graph = buildGraph(edges);
+    while (queue.size() > 0) {
+        // Remove node
+        const [node] = queue.dequeue();
+
+        // Process node
+
+        // Get children
+        const children = graph[node];
+        for (const child of children) {
+            const [childNode] = child;
+            if (visited.has(childNode)) continue;
+
+            visited.add(childNode);
+            queue.enqueue([childNode]);
+        }
+    }
+};
+
+const countComponents = (nodes, edges) => {
+    const visited = new Set();
+    let connectedComponents = 0;
```

Comment:
The name `connectedComponents` leads me to believe that this variable literally stores the components themselves (so maybe an array where each element is an object storing data about a particular component). That's not at all what this is. To make your code as clear and readable as possible, you want to show what this contains. Something like `numConnectedComponents` or even `numComponents` should work.